lib/superset/dashboard/datasets/list.rb

```diff
@@ -36,13 +37,34 @@ def schemas
         end
 
         def datasets_details
-          result.map do |details|
+          chart_datasets = result.map do |details|
             details.slice('id', 'datasource_name', 'schema', 'sql').merge('database' => details['database'].slice('id', 'name', 'backend')).with_indifferent_access
           end
+          return chart_datasets unless include_filter_datasets
+          chart_dataset_ids = chart_datasets.map{|d| d['id'] }
+          filter_dataset_ids_not_used_in_charts = filter_dataset_ids - chart_dataset_ids
+          return chart_datasets if filter_dataset_ids_not_used_in_charts.empty?
+          dashboard_datasets = chart_datasets + filter_datasets(filter_dataset_ids_not_used_in_charts)
```

Comment:
@soundarya-mv 
no real need to have another variable here, its not used any where
I would say use a comment if you want to be obvious that we are returning chart + filter datasets.

```
   ...
   return chart_datasets if filter_dataset_ids_not_used_in_charts.empty?
   
   # returning chart and filter datasets
   chart_datasets + filter_datasets(filter_dataset_ids_not_used_in_charts)
end

Reply:
I also thought of removing it. Looks like I missed it. Will update it.

Reply:
Have updated it now